app/src/main/java/zapsolutions/zap/util/TorUtil.java

```diff
@@ -0,0 +1,58 @@
+package zapsolutions.zap.util;
+
+import android.app.Activity;
+import android.app.AlertDialog;
+import android.content.DialogInterface;
+import android.content.Intent;
+
+import info.guardianproject.netcipher.proxy.OrbotHelper;
+import zapsolutions.zap.R;
+import zapsolutions.zap.connection.establishConnectionToLnd.LndConnection;
+
+public class TorUtil {
+
+    private final Activity mActivity;
+    private OnOrbotInstallConfirmedListener mOnOrbotInstallConfirmedListener;
+
+    public TorUtil(Activity activity) {
```

Comment:
I would rather not include the `Activity` inside this class.
This would let us create a Singleton and we will not take the risk of leaking the `Activity`.
I would just let the caller decide what do do if `Orbot` is not installed.
e.g. in our case let `HomeActivity` decide to show an `Dialog` .